spec/views/users/two_factor_authentication/show.html.slim_spec.rb

```diff
@@ -3,35 +3,36 @@
 describe 'users/two_factor_authentication/show.html.slim' do
   context 'user has a phone' do
     let(:user) { build_stubbed(:user, :signed_up) }
+    let(:presenter_data) { attributes_for(:generic_otp_presenter) }
 
     before do
       @otp_delivery_selection_form = OtpDeliverySelectionForm.new
       allow(view).to receive(:reauthn?).and_return(false)
       allow(view).to receive(:current_user).and_return(user)
+
+      otp_data = presenter_data.merge(
+        delivery_method: nil,
+        phone_number: view.current_user.phone
+      )
+
+      @presenter = TwoFactorAuthCode::OtpDeliveryPresenter.new(otp_data)
+
+      render
     end
 
     it_behaves_like 'an otp form'
 
     it 'has a localized heading' do
-      render
-
-      expect(rendered).to have_content t('headings.choose_otp_delivery')
+      expect(rendered).to have_content @presenter.header
```

Comment:
If we add a unit test for `OtpDeliveryPresenter` that verifies the actual expected text for `header`, then checking `@presenter.header` here is fine. Otherwise, without the unit test, this expectation would pass even if the class modifies the `header` method to something unexpected.